src/FileCache.php

```diff
@@ -46,6 +46,10 @@ class FileCache implements CacheInterface, CounterInterface
      */
     public function __construct($cache_path, $default_ttl)
     {
+        if (! file_exists($cache_path) && file_exists(dirname($cache_path))) {
+            @mkdir($cache_path, 0777, true); // ensure that the parent path exists
```

Comment:
no i don't think that we need to the creation of nested directories. 
we just need to create the cache parent folder.